diff --git a/dist/index.js b/dist/index.js
index e6c0383266ef4834c81797d7a3cd0d8ad5039d64..19045d809ed8b56cd14eab1bbfb9dbf9d746cc02 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -2243,12 +2243,18 @@ var Agent = class extends BaseResource {
     return response;
   }
   async network(messages, params) {
+    const processedParams = {
+      ...params,
+      messages,
+      requestContext: parseClientRequestContext(params.requestContext),
+      structuredOutput: params.structuredOutput ? {
+        ...params.structuredOutput,
+        schema: zodToJsonSchema2(params.structuredOutput.schema)
+      } : void 0
+    };
     const response = await this.request(`/agents/${this.agentId}/network`, {
       method: "POST",
-      body: {
-        messages,
-        ...params
-      },
+      body: processedParams,
       stream: true
     });
     if (!response.body) {
